### PR TITLE
add executable example/tutorial

### LINKED ISSUE #6: Quarto/Jupyter notebook example
On the topic of documentation. I still think we should also add an executable example of the rule as a Quatro or Juypter notebook. This is worth a discussion. I've viewed this as not unit testing but a 'how to' documentation. 

We generally try to follow the approach of the [Divio](https://documentation.divio.com) documentation approach. I think the executable examples would go high up in the file structure or be incorporated into the individual rule md files. (covert the `.md` files to, say, a `.ipynb` or `.qmd`. Discuss?

_Originally posted by @DougManuel in https://github.com/Big-Life-Lab/PHES-ODM-Validation/issues/5#issuecomment-1212964862_

### PR DESCRIPTION
**Pull request by zargot**
fixes #6

### CODE REVIEW

**Review by DougManuel — COMMENTED** [review]
Looks good. 

Consider starting with a minimal example like the text below, and then go into more complex examples such as changing directories and prettier print. 

I'd probably put the 'generating a schema' into a different file that would focus just on that. Not sure.  

```
samples = utils.import_dataset(join("assets/v2/samples-invalid.csv"))
data = {"samples": samples}
errors = validate_data(schema, data)
```

**Comment by DougManuel** [comment]
Just a small issue. The convention for the ODM is `[table]_[part]` not  `[table].[part]`. We do that because using `_` generates a valid name in all programming languages. (`.`) does not.

**Review by DougManuel — APPROVED** [review]

**Review by yulric — CHANGES_REQUESTED** [review]
Looks good, minor housekeeping points.

* Add setup instructions for the package
	* Installing packages in requirements.txt
* Think we should move the qmd file and the tutorial specific assets into its own folder. Don't want the root to get too cluttered.

File: tutorial.qmd [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+Our sample dataset has been validated successfully with no errors.
+
+Now let's see what happens when validating invalid data. We'll use the same
+schema, but replace the previous dataset with the invalid variant.
+
+```{python}
+samples = utils.import_dataset(join(ASSET_DIR, "samples-invalid.csv"))
+data = {"samples": samples}
+errors = validate_data(schema, data)
```
[the remark this answers is not in the document]
Comment (zargot) [reply tutorial.qmd]: fixed

File: tutorial.qmd [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+for e in errors:
+    print("- " + e["message"])
+```
+
+The error message starts with the name of the rule that has been broken,
+followed by the table name, field name, and row number. The actual format of
+the message varies from rule to rule, but the general idea is the same.
+
+To find out the requirements for each rule, you can read the specification in
+`/validation-rules/<rule_name>.md`.
```
[the remark this answers is not in the document]
Comment (zargot) [reply tutorial.qmd]: fixed

**Review by yulric — APPROVED** [review]

**Pull request merged**